src/module/rules/rule-element/token-marker/rule-element.ts

```diff
@@ -0,0 +1,68 @@
+import { TokenDocumentPF2e } from "@scene/index.ts";
+import { SlugField } from "@system/schema-data-fields.ts";
+import { ErrorPF2e } from "@util";
+import { UUIDUtils } from "@util/uuid.ts";
+import type { StringField } from "types/foundry/common/data/fields.d.ts";
+import { RuleElementPF2e, RuleElementSchema, RuleElementSource } from "../index.ts";
+import { MarkTargetPrompt } from "./prompt.ts";
+
+/** Remember a token for later referencing */
+class TokenMarkerRuleElement extends RuleElementPF2e<TokenMarkSchema> {
+    static override defineSchema(): TokenMarkSchema {
+        const { fields } = foundry.data;
+        return {
+            ...super.defineSchema(),
+            slug: new SlugField({ required: true, nullable: false, initial: undefined }),
+            uuid: new fields.StringField({ required: false, nullable: true, initial: null }),
+        };
+    }
+
+    override async preCreate({ ruleSource, itemSource, pendingItems }: RuleElementPF2e.PreCreateParams): Promise<void> {
+        if (this.ignored) return;
+
+        this.uuid &&= this.resolveInjectedProperties(this.uuid);
+
+        const token =
+            fromUuidSync(this.uuid ?? "") ??
+            (game.user.targets.size === 1
+                ? Array.from(game.user.targets)[0]!.document
+                : await new MarkTargetPrompt({ prompt: null, requirements: null }).resolveTarget());
+        if (!(token instanceof TokenDocumentPF2e)) {
+            // No token was targeted: abort creating item
+            pendingItems.splice(pendingItems.indexOf(itemSource), 1);
+            return;
+        }
+
+        this.#checkRuleSource(ruleSource);
+        this.uuid = ruleSource.uuid = token.uuid;
+        itemSource.name = `${itemSource.name} (${token.name})`;
```

Comment:
Does `token.name` respect the `tokenSetsNameVisibility` setting?

Reply:
I removed it and will follow up with some visual indication on tokens.